src/main/java/io/stargate/sgv2/jsonapi/service/schema/collections/CollectionSchemaObject.java

```diff
@@ -406,19 +405,24 @@ public static CreateCollectionCommand collectionSettingToCreateCollectionCommand
             lexicalConfig.enabled(), lexicalConfig.analyzerDefinition());
 
     // construct the CreateCollectionCommand.options.rerankingConfig
-    CollectionRerankingConfig rerankingConfig = collectionSetting.rerankingConfig;
-    CreateCollectionCommand.Options.RerankingServiceConfig rerankingServiceConfig = null;
+    CollectionRerankDef rerankingConfig = collectionSetting.rerankingConfig;
+    CreateCollectionCommand.Options.RerankServiceDesc rerankServiceDesc = null;
     if (rerankingConfig.enabled()) {
-      rerankingServiceConfig =
-          new CreateCollectionCommand.Options.RerankingServiceConfig(
+      if (rerankingConfig.rerankingProviderConfig() == null) {
+        throw new IllegalStateException(
+            "Reranking is enabled but rerankingProviderConfig is empty. The comment in the collection '%s' has some problems."
+                .formatted(collectionSetting.name().table()));
+      }
+      rerankServiceDesc =
```

Comment:
in the Tables schema, we had the *Def objects create their *Desc pair, e.g CollectionRerankDef.getRerankDesc()